streamer/src/packet.rs

```diff
@@ -28,8 +29,11 @@ pub fn recv_from(obj: &mut Packets, socket: &UdpSocket, max_wait_ms: u64) -> Res
             std::cmp::min(i + NUM_RCVMMSGS, PACKETS_PER_BATCH),
             Packet::default(),
         );
-        match recv_mmsg(socket, &mut obj.packets[i..]) {
-            Err(_) if i > 0 => {
+        match recv_mmsg(socket, &mut obj.packets[i..], max_wait_ms) {
+            Err(e) if i > 0 => {
+                if e.kind() == io::ErrorKind::WouldBlock {
+                    break;
```

Comment:
The design of the wait here is to coalesce packets within the wait interval which helps with some stages like banking stage to keep more transactions together. Seems like this would break up those batches more since we would exit out earlier here.

Reply:
The upper level callers are attempting to do the coalescing within a 1ms timeout. From some targeted logging I was seeing the majority of calls either filling the entire packets array (128 packets) in the first call, or just getting a handful of packets on the first call. Subsequent calls after putting the socket in non-blocking mode was resulting in >500 calls to recvmmsg within the 1ms window usually picking up only a few more packets at best. The high spinning was mostly just burning CPU zeroing buffers and taking syscall overhead. This was running on testnet.

Reply:
I see